### ISSUE TITLE
Serialization/deserialization of FollowJointTrajectory action

### ISSUE THREAD

**Issue by ted-miller** [micro-ROS/rosidl_typesupport_microxrcedds#48]
**Describe the bug**
This is a continuation of the debugging around this: https://github.com/micro-ROS/rmw_microxrcedds/issues/247
We are seeing that FJT messages are not being deserialized correctly. It seems to occur when the message exceeds some size, though I haven't isolated what the exact 'cutoff' size is. 
(I'm not sure if this issue belongs here or in MicroCDR. So, my apologies if this is the wrong repo for this.)

**To Reproduce**
I basically took the [test procedure from here](https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/blob/51c7e2b36a4ddb8175c4662145c84a66c23be317/test/c/src/test_typesupport.cpp#L94-L111) and recreated it. 

This code serializes the action-goal message, then deserializes it, and then compares to two messages.

[serialization_test.zip](https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/files/8593160/serialization_test.zip)
```
export RMW_IMPLEMENTATION=rmw_microxrcedds
source install/local_setup.bash
ros2 run rclc_examples serialization_test
```

Relevant code for the output:
```
	if (!callbacks->cdr_serialize(&instanceBefore, &mb_writer))
		Ros_Debug_BroadcastMsg("FAILED TO SERIALIZE");
	else
		Ros_Debug_BroadcastMsg("serialization OK");

	if ((mb_writer.iterator - mb_writer.init) != topic_size)
		Ros_Debug_BroadcastMsg("INVALID SERIALIZED SIZE: topic_size = %d, mb_writer size = %d", topic_size, mb_writer.iterator - mb_writer.init);

	if (callbacks->cdr_deserialize(&mb_reader, &instanceAfter))
		Ros_Debug_BroadcastMsg("FAILED TO DE-SERIALIZE");
	else
		Ros_Debug_BroadcastMsg("de-serialization OK");
```
Output:
```
ted@ted-Ubuntu:~/jta_size_testing$ ros2 run rclc_examples serialization_test
Testing CDR for FJT
size of instanceBefore = 373560
instanceBefore has been allocated
instanceBefore joint names have been set
instanceBefore point 1 data set
instanceBefore point 2 data set
instanceBefore point 3 data set
instanceBefore point 4 data set
instanceBefore point 5 data set
size of instanceAfter = 373560
instanceAfter has been allocated
function pointers obtained
topic_size = 768
ucdr_init_buffer complete
serialization OK
INVALID SERIALIZED SIZE: topic_size = 768, mb_writer size = 692
FAILED TO DE-SERIALIZE
****** joint name 0 is wrong: 
****** joint name 1 is wrong: 
****** joint name 2 is wrong: 
****** joint name 3 is wrong: 
****** joint name 4 is wrong: 
****** joint name 5 is wrong: 
****** joint name 6 is wrong: 
****** joint name 7 is wrong: 
****** joint name 8 is wrong: 
****** joint name 9 is wrong: 
****** joint name 10 is wrong: 
****** joint name 11 is wrong: 
CDR Testing complete

```
As you can see, the size of the 'writer' is not correct. Then the call to cdr_deserialize returns false.

Can you guys take a look at the FJT message to see why it won't serialize correctly?

Thank you for your support!

**Comment by Acuadros95** [comment]
Hi @ted-miller,

I replicated the issue:
- The deserialization error on joints names is fixed after setting the string sequence size: `instanceBefore.goal.trajectory.joint_names.size = 12;`
- The mismatch between the topic serialized sizes is caused by a bug on `micro_ros_utilities_get_static_size`. We will handle this error, but it should not affect the serialization/deserialization process.

**Comment by ted-miller** [comment]
Thank you for the fix to the "get size" function. 

As you mentioned, it did not affect the serialization process. So, now I'm working on injecting additional output into `msg__type_support_c.c.em` to try to determine which element(s) of the message is failing.

**Comment by ted-miller** [comment]
Hi @Acuadros95

I'm not sure why, but it seems that the deserialization is working now. I'm trying to figure out what I changed, but all I'm finding is print statements used during debugging.

I did find that I was using an Agent that was in a workspace which didn't have your get-size. Now my agent is in the workspace with your fix. 

Is there any credibility to the idea that your fix would influence the Agent behavior? (I'm at a loss for other ideas...)
